toolchain/check/import_cpp.cpp

```
@@ -540,7 +540,49 @@ static auto ImportClassObjectRepr(Context& context,
   static_assert(SemIR::CustomLayoutId::FirstFieldIndex == 2);
 
   // TODO: Import vptr(s).
-  // TODO: Import bases.
+
+  // Import bases.
+  for (const auto& base : clang_def->bases()) {
+    if (base.isVirtual()) {
+      // TODO: Handle virtual bases. We don't actually know where they go in the
+      // layout. We may also want to use a different size in the layout for
+      // `partial C`, excluding the virtual base.
+      continue;
+    }
+
+    auto [base_type_inst_id, base_type_id] =
+        MapType(context, import_ir_inst_id, base.getType());
+    auto* base_class = base.getType()->getAsCXXRecordDecl();
+    if (!base_class || !base_type_id.has_value()) {
+      // If the base class's type can't be mapped, skip it.
```

Should there be a TODO here, too? It seems odd to fail silently.

---

CHECK LG